packages/next/src/server/async-storage/request-async-storage-wrapper.ts

```diff
@@ -76,6 +98,9 @@ export const RequestAsyncStorageWrapper: AsyncStorageWrapper<
       previewProps = (renderOpts as any).previewProps
     }
 
+    const [_callback, afterContext] = wrapCallbackForAfter(renderOpts, callback)
```

Comment:
I'm wondering if it'd be better to do the wrapping around the `storage.run` line at the end instead. It feels a bit confusing to have all of this run within RAS when `afterContext.run` does things that should conceptually be outside of that.

(it's not a problem in practice because `afterContext.run` shadows RAS anyway, but maybe it'd be less convoluted that way)

Reply:
moved it out to run outside of RAS